MetricsUploader/MetricsUploaderWindows.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "MetricsUploader/MetricsUploader.h"
+#include "OrbitBase/Logging.h"
+
+namespace orbit_metrics_uploader {
+
+using orbit_client_protos::OrbitLogEvent;
+using orbit_client_protos::OrbitLogEvent_LogEventType;
+
+constexpr const wchar_t* kMetricsUploaderClientDllName = L"metrics_uploader_client.dll";
+constexpr const char* kSendLogEventFunctionName = "SendOrbitLogEvent";
+constexpr const char* kStartUploaderClientFunctionName = "StartUploaderClient";
+
+MetricsUploader::MetricsUploader() {
+  CHECK(!is_init_);
+
+  metrics_uploader_client_dll_ = LoadLibrary(kMetricsUploaderClientDllName);
+
+  if (NULL != metrics_uploader_client_dll_) {
+    LOG("Metrcs uploader client library is loaded");
+
+    send_log_event_addr_ = (void (*)(const uint8_t*, int))GetProcAddress(
+        metrics_uploader_client_dll_, kSendLogEventFunctionName);
+    if (NULL == send_log_event_addr_) {
+      LOG("%s function not found", kSendLogEventFunctionName);
+      FreeLibrary(metrics_uploader_client_dll_);
+    }
+
+    void (*start_uploader_client_addr_)();
+    start_uploader_client_addr_ =
+        (void (*)())GetProcAddress(metrics_uploader_client_dll_, kStartUploaderClientFunctionName);
+    if (NULL == start_uploader_client_addr_) {
+      LOG("%s function not found", kStartUploaderClientFunctionName);
+      FreeLibrary(metrics_uploader_client_dll_);
+    }
+
+    // set up connection and create a client
+    start_uploader_client_addr_();
+    is_init_ = true;
+  } else {
+    LOG("Metrcs uploader client library is not found");
+  }
+}
+
+MetricsUploader::~MetricsUploader() {
+  if (is_init_) {
+    FreeLibrary(metrics_uploader_client_dll_);
+  }
+}
+
+void MetricsUploader::SendLogEvent(OrbitLogEvent_LogEventType log_event_type) {
+  if (is_init_) {
```

Comment:
Nit: `if (send_log_event_addr_ != nullptr)` instead

That's basically the same condition but easier to understand for someone just looking at this function.

Reply:
Changed the condition to `if (send_log_event_addr_ != nullptr)`.